source/renderer.js

```diff
@@ -1,39 +1,50 @@
 'use strict';
-const electron = require('electron');
+const electron = window.require('electron');
 const util = require('./util');
 
 const {ipcRenderer: ipc} = electron;
 
 ipc.callMain = (channel, data) => new Promise((resolve, reject) => {
 	const {sendChannel, dataChannel, errorChannel} = util.getResponseChannels(channel);
+	const id = `${Date.now()}.${Math.random()}`;
+	const uniqueDataChannel = `${dataChannel}-${id}`;
+	const uniqueErrorChannel = `${errorChannel}-${id}`;
```

Comment:
This is the same as the one in `main.js`. You could move them to `util.js` as functions that return unique channels.